src/NHibernate/Cfg/Environment.cs

```diff
@@ -78,6 +79,7 @@ public static string Version
 		public const string ConnectionStringName = "connection.connection_string_name";
 
 		// Unused, Java-specific
+		// But has many code usage though.
 		public const string SessionFactoryName = "session_factory_name";
```

Comment:
Not switched to obsolete because its many code usages imply a lot of changes, and so I am not sure it has really not any use. (And I do not intend to investigate further the subject.)